src/containers/Main/Faucet.tsx

```diff
@@ -92,48 +91,44 @@ const ButtonWrapper = styled.div`
 `;
 
 type HOCProps = FormComponentProps & RouteComponentProps;
-interface FaucetProps extends HOCProps {
-  getFromFaucet: $TSFixMe;
-}
+type FaucetProps = HOCProps;
 
-function Faucet({ form, getFromFaucet }: FaucetProps) {
+function Faucet({ form }: FaucetProps) {
   const { getFieldDecorator } = form;
-  const [isLoading, setIsLoading] = useState(false);
+
+  const {
+    mutate: requestFaucetFunds,
+    isLoading: isRequestFaucetFundsLoading,
+  } = useRequestFaucetFunds({
+    onSuccess: (_data, variables) => {
```

Comment:
What about defining this config in `hooks/useRequestFaucetFunds`?

Reply:
I wouldn't. In order to make the hooks responsible for fetching data as flexible as possible, I'd only make them deal with the actual HTTP request and handling the data that comes from it.
Any behaviour that's specific to the component rendering the hook (in this case displaying a toast on success and failure) should be coded inside the hook call itself. This way if we want to reuse the hook elsewhere, but make the request fail silently or do something else on success, it's very easy to just write that logic in the relevant callbacks (`onSuccess`, `onError`, `onSettled` etc.).

Reply:
That makes sense, although I would posit that a single request would get handled similarly throughout the app. I think it will be nicer with this defined in the "container". As it stands now it seems like a lot for just accessing data in a presentational component

Reply:
My mistake was not refactoring the `Faucet` completely in this PR. It will make more sense once I'm done with the refactoring, then we can talk about this further!